src/signed-api-fetch/data-fetcher.ts

```diff
@@ -42,12 +57,12 @@ const callSignedDataApi = async (url: string): Promise<SignedData[]> => {
     }
   );
 
-  // TODO: Pass the response error body (if available). See OEV repo for details.
-  if (!result.success) {
-    throw new Error([`HTTP call failed: `, url, result.error].join('\n'));
```

Comment:
In general, we should not rethrow errors because it:
1. Messes up with the stack trace (using Error.cause mitigates this though)
2. The point of catching the errors is likely to process it somehow. If we process an error and re-throw it will likely be handled twice which is not good from design perspective.